src/main/java/com/coveo/pushapiclient/ApiCore.java

```diff
@@ -9,19 +9,48 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-// TODO: LENS-934 - Support throttling
 class ApiCore {
   private final HttpClient httpClient;
   private final Logger logger;
+  private final BackoffOptions options;
 
   public ApiCore() {
-    this.httpClient = HttpClient.newHttpClient();
-    this.logger = LogManager.getLogger(ApiCore.class);
+    this(HttpClient.newHttpClient(), LogManager.getLogger(ApiCore.class));
   }
 
   public ApiCore(HttpClient httpClient, Logger logger) {
+    this(httpClient, logger, new BackoffOptionsBuilder().build());
+  }
+
+  public ApiCore(HttpClient httpClient, Logger logger, BackoffOptions options) {
     this.httpClient = httpClient;
     this.logger = logger;
+    this.options = options;
+  }
+
+  public HttpResponse<String> callApiWithRetries(HttpRequest request)
+      throws IOException, InterruptedException {
+    int nbRetries = 0;
+    long delayInMilliseconds = this.options.getRetryAfter();
+
+    while (true) {
+      String uri = request.uri().toString();
+      String reqMethod = request.method();
+      this.logger.debug(reqMethod + " " + uri);
+      HttpResponse<String> response =
+          this.httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+      this.logResponse(response);
+
+      if (response != null
+          && response.statusCode() == 429
+          && nbRetries < this.options.getMaxRetries()) {
+        Thread.sleep(delayInMilliseconds);
```

Comment:
As discussed live, we should look into something a bit more sophisticated than Thread.sleep, as those could hog resources, especially since we might end up sleeping for a long time.

Reply:
Discussed live but commenting here for visibility:
Agree 100%, so I decided to use Resilience4j (which appears to be popular/well-maintained) to implement a more solid retry mechanism with jitter:
https://github.com/resilience4j/resilience4j